### DISCUSSION TITLE
Bind DatagridSelectColumn to an Enum Field (cast error)

### DISCUSSION THREAD

**Discussion by michele-bettin** [Megabit/Blazorise#5069] [General]
Hi guys! I prepared a quick demo to explain my problem to you.
I have a data model that has an Enum for the Role properties (not a string, because that would make my app inconsistent). When I want to modify a Employee with my dropdown blasorise is unable to understand that the Value of the select is an enumeration and cast but it gives me an error telling me that it is unable to convert String to UsersRoles.
Does anyone have any suggestions on how to do this? 

```
<DataGrid TItem="Employee" Data="@employeeList" PageSize="5" Responsive Editable Filterable EditMode="DataGridEditMode.Popup">
    <DataGridSelectColumn TItem="Employee"  Field="@nameof( Employee.Role )" Caption="Gender" Editable
                          Data="RolesNamesSource" ValueField="(x) => ((UsersRoles)Enum.Parse(typeof(UsersRoles), (string)x))" TextField="(x) => (string)x" />
    <DataGridCommandColumn />
</DataGrid>

@code {

    private List<Employee> employeeList = new List<Employee>() { new Employee(UsersRoles.Standard), new Employee(UsersRoles.Premium), new Employee(UsersRoles.Admin) };
    private List<string> RolesNamesSource = Enum.GetNames(typeof(UsersRoles)).ToList();

    protected override void OnInitialized()
    {
        
        base.OnInitialized();

    }

    class Employee
    {
        public Employee(UsersRoles role)
        {
            Role = role;
        }

        public UsersRoles Role { get; set; }
    }

    enum UsersRoles
    {
        Standard,
        Premium,
        Admin
    }

}
```

**Comment by David-Moreira** [comment]
Hello @michele-bettin 
The serialized value (the value string) might not be what you are expecting.
Could you please read on the bit about enums here:
https://blazorise.com/blog/how-to-handle-select-with-primitive-and-complex-types
And see if it helps you?

Thanks!

**Reply by michele-bettin** [reply]
Hello @David-Moreira, the article that you sent to me rely on the concept of TValue which specify the value that the Select component is handling (UsersRoles enum in my case). Here the problem is that in DataGridSelectColumn there is no TValue attribute...

**Reply by David-Moreira** [reply]
Hello @michele-bettin 
Even though there's no explicit `TValue`, behind the scenes it will render a `Select` with the defined `TValue` according to 
`ValueField="(x) => ((UsersRoles)Enum.Parse(typeof(UsersRoles)`. 

Have you tried applying the json converter option that's in the blog post?

**Reply by michele-bettin** [reply]
@David-Moreira you can try yourself on the code I provided (maybe I'am missing something...) but the same conversion error (impossible convert string to UsersRoles) is displayed...

**Reply by David-Moreira** [reply]
Please follow on #5073

**Reply by michele-bettin** [reply]
@David-Moreira do you report my problem as a possible bug?

**Reply by David-Moreira** [reply]
Yes, I need to check the example you gave us and why would it work differently from the blog post, according to the information you've provided us.

Thanks.

**Reply by michele-bettin** [reply]
@David-Moreira I think the problem is that i'm not working with a Select but with a Datagrid and even if under the hood a Select component is used something in the middle probably goes wrong...
However... If a bug has been solved how can I retrive the last (but probably unstable) version of Blasorise If I installed it in my project with Nuget... 
Sorry for the questions but I'm new but I would like to collaborate with you...

**Reply by David-Moreira** [reply]
I will let you know once I take a look at it and if a fix was needed.

We release new patches according to the urgency of the fixes and with some cadence, generally 1-2 weeks at least. You can just update your nuget once we release the patch.

We currently do not release unstable or nightly versions.

**Reply by David-Moreira** [reply]
**PS:** If you really need a workaround in the time being, you can use a regular `DataGridColumn` and use the DisplayTemplate/EditTemplate to create the `Select` as you wish.
